// react-native-datefield/lib/commonjs/Input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const defaultProps = {
  maxLength: 2,
  returnKeyType: 'next'
};

const Input = /*#__PURE__*/_react.default.forwardRef(({
  style,
  ...attributes
}, ref) => {
  return /*#__PURE__*/_react.default.createElement(_reactNative.TextInput, _extends({}, attributes, {
    ref: ref,
    blurOnSubmit: false,
    keyboardType: "numeric",
    style: [styles.input, style]
  }));
});

const styles = _reactNative.StyleSheet.create({
  input: {
    textAlign: 'center'
  }
});

const areEqual = (prevProps, nextProps) => {
  return prevProps.value === nextProps.value;
};

Input.defaultProps = defaultProps;

var _default = /*#__PURE__*/_react.default.memo(Input, areEqual);

exports.default = _default;
